swarm/src/either.rs

```diff
@@ -0,0 +1,248 @@
+// Copyright 2021 Protocol Labs.
```

Comment:
In my eyes both `Toggle` and the implementation of `NetworkBehaviour` on `Either` should live in `swarm/src/behaviour/`. What do you think @mexskican? In case you agree, would you mind preparing an additional pull request moving `Toggle`, which we can merge before this pull request, and in addition move `either.rs` in this pull request?

Reply:
I agree, it will be better organized.

- I created a [PR](https://github.com/libp2p/rust-libp2p/pull/2375) to move `Toggle` into `swarm::behaviour`.
- Once the above PR is merged I will update this one and move `either.rs` into the `behaviour` module.

Reply:
Done